### ISSUE TITLE
plugin/ecs: support blue/green deployments

### ISSUE THREAD

**Issue by richstokes** [hashicorp/waypoint#2163]
Instead of doing a regular cutover, which seemed to result in downtime when testing, it would be helpful if waypoint could use CodeDeploy to do a blue/green deployment.

**Comment by evanphx** [comment]
@richstokes Seems like to use those features, the user would have to setup all the CodeDeploy bits as well. CodeDeploy is an orchestration engine, a bit like waypoint. We're a bit unsure how we'd enable that without a lot of confusion.

On the topic of zero-downtime deployments though, Waypoint could (and should) do more about how it switches to new deployments to make sure there is no downtime. I'll open an issue about that.

**Comment by evanphx** [comment]
I've created #2187 to track the broader issue.

**Comment by richstokes** [comment]
Yep, all makes sense. The reason I created this was because the current deployment method seems to have downtime, which is odd/unexpected because the default ECS deployment strategy is to do a kind of rolling deployment where there is no downtime. 

I think this is because Waypoint seems to create a new service (instead of letting ECS deploy a new _version_ of a service) and orchestrates the cutover manually, causing some downtime.
